Source/com/drew/imaging/jpeg/JpegMetadataReader.java

```diff
@@ -71,45 +72,80 @@
     );
 
     @NotNull
-    public static Metadata readMetadata(@NotNull InputStream inputStream, @Nullable Iterable<JpegSegmentMetadataReader> readers) throws JpegProcessingException, IOException
+    public static Metadata readMetadata(@NotNull InputStream inputStream, @Nullable Iterable<JpegSegmentMetadataReader> readers, @NotNull MetadataContext context) throws JpegProcessingException, IOException
     {
         Metadata metadata = new Metadata();
-        process(metadata, inputStream, readers);
+        process(metadata, inputStream, readers, context);
         return metadata;
     }
 
+    @NotNull
+    public static Metadata readMetadata(@NotNull InputStream inputStream, @Nullable Iterable<JpegSegmentMetadataReader> readers) throws JpegProcessingException, IOException
+    {
+        return readMetadata(inputStream, readers, null);
+    }
+
+    @NotNull
+    public static Metadata readMetadata(@NotNull InputStream inputStream, @NotNull MetadataContext context) throws JpegProcessingException, IOException
+    {
+        return readMetadata(inputStream, null, context);
+    }
+
     @NotNull
     public static Metadata readMetadata(@NotNull InputStream inputStream) throws JpegProcessingException, IOException
     {
-        return readMetadata(inputStream, null);
+        // TODO document this default context?
```

Comment:
As explained in a comment: I've added this comment throughout to mark where `new MetadataContext()` is called, which is done mostly for backwards compatibility.